### PR TITLE
FIX : always return array in pick_types

### PR DESCRIPTION
**Pull request by agramfort**

### CODE REVIEW

**Comment by dengemann** [comment]
Finally! Good move

**Comment by agramfort** [comment]
I got bitten by this with students yesterday.

but I need to fix some tests...

**Comment by dengemann** [comment]
> I got bitten by this with students yesterday.

bitten by students?

> but I need to fix some tests...

Ok, let me know. I suspect it might have some funky impact on some of our pipelines but I'm happy to live with that as it will remove lines to deal with this unfavorable API design.

File: mne/io/pick.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def pick_types(info, meg=True, eeg=False, stim=False, eog=False, ecg=False,
     myinclude += include
 
     if len(myinclude) == 0:
-        sel = []
+        sel = np.array([])
```
Comment (Eric89GXL) [inline mne/io/pick.py]: needs to be `np.array([], int)`

Comment (Eric89GXL) [inline mne/io/pick.py]: (I will push)

**Comment by Eric89GXL** [comment]
Making the change fixed at least one of the broken tests (test_dipole.py) for me, I assume it will fix the others since they all were of the form "must be boolean array or array of int". Pushed.

**Comment by agramfort** [comment]
thanks @Eric89GXL for taking over

**Comment by Eric89GXL** [comment]
Thanks for the fix @agramfort

**Pull request merged**